include/api.php

```diff
@@ -3276,72 +3276,706 @@ function api_oauth_access_token($type) {
+
+		if ($filetype = "image/jpeg")
+			$fileext = "jpg";
+		else if ($filetype = "image/png")
+			$fileext = "png";
+
+		// change specified profile or all profiles to the new resource-id
+		if($is_default_profile) {
+			$r = q("UPDATE `photo` SET `profile` = 0 WHERE `profile` = 1 AND `resource-id` != '%s' AND `uid` = %d",
+				dbesc($data['photo']['id']),
+				intval(local_user())
+			);
+
+			$r = q("UPDATE `contact` SET `photo` = '%s', `thumb` = '%s', `micro` = '%s'  WHERE `self` AND `uid` = %d",
+				dbesc(App::get_baseurl() . '/photo/' . $data['photo']['id'] . '-4.' . $fileext),
+				dbesc(App::get_baseurl() . '/photo/' . $data['photo']['id'] . '-5.' . $fileext),
+				dbesc(App::get_baseurl() . '/photo/' . $data['photo']['id'] . '-6.' . $fileext),
+				intval(local_user())
+			);
+		} else {
+			$r = q("update profile set photo = '%s', thumb = '%s' where id = %d and uid = %d",
+				dbesc(App::get_baseurl() . '/photo/' . $data['photo']['id'] . '-4.' . $filetype),
+				dbesc(App::get_baseurl() . '/photo/' . $data['photo']['id'] . '-5.' . $filetype),
+				intval($_REQUEST['profile']),
+				intval(local_user())
+					);
+		}
+
+		// we'll set the updated profile-photo timestamp even if it isn't the default profile,
+		// so that browsers will do a cache update unconditionally
+
+		$r = q("UPDATE `contact` SET `avatar-date` = '%s' WHERE `self` = 1 AND `uid` = %d",
+			dbesc(datetime_convert()),
+			intval(local_user())
+		);
+
+		// Update global directory in background
+		//$user = api_get_user(get_app());
+		$url = App::get_baseurl() . '/profile/' . get_app()->user['nickname'];
+		if ($url && strlen(get_config('system','directory'))) {
+			proc_run(PRIORITY_LOW, "include/directory.php", $url);
+		}
+
+		require_once('include/profile_update.php');
+		profile_change();
+
+		// output for client
+		if ($data) {
+			return api_account_verify_credentials($type);
+		}
+		else {
+			// SaveMediaToDatabase failed for some reason
+			throw new InternalServerErrorException("image upload failed");
+		}
+	}
+
+	// place api-register for photoalbum calls before 'api/friendica/photo', otherwise this function is never reached
+	api_register_func('api/friendica/photoalbum/delete', 'api_fr_photoalbum_delete', true, API_METHOD_DELETE);
+	api_register_func('api/friendica/photoalbum/update', 'api_fr_photoalbum_update', true, API_METHOD_POST);
+	api_register_func('api/friendica/photos/list', 'api_fr_photos_list', true);
+	api_register_func('api/friendica/photo/create', 'api_fr_photo_create_update', true, API_METHOD_POST);
+	api_register_func('api/friendica/photo/update', 'api_fr_photo_create_update', true, API_METHOD_POST);
+	api_register_func('api/friendica/photo/delete', 'api_fr_photo_delete', true, API_METHOD_DELETE);
+	api_register_func('api/friendica/photo', 'api_fr_photo_detail', true);
+	api_register_func('api/account/update_profile_image', 'api_account_update_profile_image', true, API_METHOD_POST);
+
+
+	function CheckACLInput($acl_string) {
```

Comment:
Standards: Please name this function with the most prevalent convention, which is `lower_case` for global functions.